sci-electronics/splat/splat-1.4.2.ebuild

```diff
@@ -44,17 +46,17 @@ src_configure() {
 src_compile() {
 	local CC=$(tc-getCC) CXX=$(tc-getCXX)
 
-	${CXX} -Wall ${CXXFLAGS} ${LDFLAGS} itwom3.0.cpp splat.cpp -o rfsplat -lm -lbz2 || die
+	${CXX} -Wall ${CXXFLAGS} ${LDFLAGS} itwom3.0.cpp splat.cpp -o rfsplat -lm -lbz2
 	if use hires; then
-		cp "${S}/hires.h" "${S}/splat.h"
-		${CXX} -Wall ${CXXFLAGS} ${LDFLAGS} itwom3.0.cpp splat.cpp -o rfsplat-hd -lm -lbz2 || die
+		cp "${S}"/{hires,splat}.h || die
+		${CXX} -Wall ${CXXFLAGS} ${LDFLAGS} itwom3.0.cpp splat.cpp -o rfsplat-hd -lm -lbz2
 	fi
 
-	cd utils
+	cd utils || die
+
 	${CC} -Wall ${CFLAGS} ${LDFLAGS} citydecoder.c -o citydecoder
```

Comment:
CPPFLAGS, dies?

Reply:
Personally I'd just use make implicit rules for these, e.g.
```
emake citydecoder usgs2sdf
emake LDLIBS=-lm bearing
etc...
```
Will respect all of the *FLAGS, and can use normal `append-*flags` to add more. Rather than local CC= use `tc-export CC` though.

obv no `|| die` needed with emake, and it'll also echo the commands in the build log

Reply:
In case wasn't clear, that works without a Makefile

Reply:
(That's a far nicer suggestion, and agreed re overriding CC)

Reply:
Not useful here but may be good to mention you can even do easy rules for multiple object files, e.g. with main.c src1.c src2.c src3.c, e.g.
```
emake -f /dev/null -E "main: src1.o src2.o src3.o"
```
/dev/null would ignore a pre-existing Makefile if it was there (like something really bad/unusable that's hard to patch), and it'd build a binary named "main" that you can newbin into its proper name

If really must call ${CC} directly, I'd use an array or `set` to be able to echo the command at least, but generally I can't think of many cases should do that over using emake

Reply:
@ionenwks cool, thank you for writing that up!

Reply:
...should look a bit better now :)